### ISSUE TITLE
Support SSL for v0.8.3

### ISSUE THREAD

**Issue by joestein** [Shopify/sarama#154]
Kafka is adding SSL support => https://issues.apache.org/jira/browse/KAFKA-1477 and would be great to have this available in go libraries.

**Comment by uovobw** [comment]
any news on this? i am planning on exposing a kafka cluster to the net securing it with stunnel and client side certificates. if there already is some code i can contribute to that...

**Comment by eapache** [comment]
Sarama's support is ready to go... we're waiting on upstream kafka to release this in the broker

**Comment by uovobw** [comment]
i heard that is scheduled to happen in version 0.9, that might be a long way to come yet. 
is there any chance that, with external ssl-termination as i wrote above, this could be pushed earlier?

**Comment by eapache** [comment]
#156 needs to be rebased and have any merge conflicts resolved, but it should be all you need.

**Comment by uovobw** [comment]
awesome, i'll get around to it if i get the time. thanks

**Comment by uovobw** [comment]
@eapache i have submitted a new pr https://github.com/Shopify/sarama/pull/493 that rebases this change to master (banch support-ssl). tests pass, let me know how i can help

**Comment by eapache** [comment]
Implemented (for now) by #493
